contrib/postgres_proxy/filters/network/source/postgres_message.h

```diff
@@ -350,6 +350,11 @@ template <typename... Types> class MessageImpl : public Message, public Sequence
     uint64_t pos = start_pos;
     uint64_t left = length;
     validation_result_ = Sequence<Types...>::validate(data, start_pos, pos, left);
+    if (validation_result_ != Message::ValidationOK) {
+      return validation_result_;
+    }
+    // verify that parser iterated over entire message and nothing is left.
+    validation_result_ = left == 0 ? Message::ValidationOK : Message::ValidationFailed;
```

Comment:
Seems like this would be both clearer here, and perhaps clearer in coverage that it's not covered by the new test (unless it's covered by an existing test), as an if rather than a ternary. Also shorter, though more lines.
```
if (left != 0) {
  return Message::ValidationFailed;
}
```

Reply:
Good point with coverage. Thanks!

Reply:
Done.